### PR TITLE
Task 4 - in progress

### PR DESCRIPTION
**Pull request by kimmimy**
PR -https://github.com/kimmimy/nodejs-aws-fe/pull/4
Cloud front - d2xf5mqsbfjelp.cloudfront.net

4.1 - done
4.2 - done
4.3 - done

createProduct handler can be tested via postman as openAPI schema for this handler hasn't created yet
data
{
"title": "title",
"description": string,
"count": number,
"price": number
}

URLS
GET - https://0q6hxa3bf2.execute-api.eu-west-1.amazonaws.com/dev/products
GET - https://0q6hxa3bf2.execute-api.eu-west-1.amazonaws.com/dev/products/{id}
POST - https://0q6hxa3bf2.execute-api.eu-west-1.amazonaws.com/dev/products

POST/products lambda functions returns error 400 status code if product data is invalid - done

All lambdas do console.log for each incoming requests and their argument - done

All lambdas return error 500 status code on any error (DB connection, any unhandled error in code) - done

Transaction based creation of product - done

### CODE REVIEW

**Comment by kurk28** [comment]
It's looks like your commands from .sql will not works. Am I right? You can start it in one transaction.
For validation you could use Joi (my recommendation)
And it will be great if you start to use separated log file. In the future, you may decide to use some logging app and you have to change all handlers, If you will use separated log file you could change only one file.

Important: I don't see where do you connect to BD. And where is handler for create product? Also on post request without body I get all products back.

P.S. центрИфигу

**Comment by kimmimy** [comment]
> It's looks like your commands from .sql will not works. Am I right? You can start it in one transaction.
> For validation you could use Joi (my recommendation)
> And it will be great if you start to use separated log file. In the future, you may decide to use some logging app and you have to change all handlers, If you will use separated log file you could change only one file.
> 
> Important: I don't see where do you connect to BD. And where is handler for create product? Also on post request without body I get all products back.
> 
> P.S. центрИфигу



I don't use queries from .sql file here in project. I used them in the DBeaver script to create and fill tabels, so just saved them in SQL file. if case I need to create it again, I will not write queries and data again.

As for validation, thanks. But we don't have such a requirement for now. So, will be implemented later, when I have free time.


here is my connection.

product-service/src/db/databaseService.ts

  const client = new Client({
    user: process.env.DB_USER,
    host: process.env.DB_HOST,
    database: process.env.DB_NAME,
    password: process.env.DB_PASSWORD,
    port: +process.env.DB_PORT
  });
  
    await client.connect();
    
    
    here is createProduct handler.
    product-service/src/functions/createProduct/lambdaFunction.ts
    
    
    As you can notice in the create product handler there is no opportunity to get all products from this function.

**Review by artem-blazhko — APPROVED** [review]

File: product-service/src/functions/createProduct/lambdaFunction.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      }, 201);
+    } else {
+      return formatJSONResponse({
+        data: `Validation Error:: ${JSON.stringify(event.body)}`
+      }, 400);
+    }
+  } catch (error) {
+    console.log('Error::', error);
+
+    formatJSONError(error || "something went wrong", 404);
```
Comment (kektour) [inline product-service/src/functions/createProduct/lambdaFunction.ts:31]:
```suggestion
    return formatJSONError(error || "something went wrong", 404);
```

**Comment by kektour** [comment]
You have commited your dist folder. Add rule to `.gitignore`

**Comment by kektour** [comment]
Please provide link to FE cloudfront app

**Pull request merged**